hyp3lib/get_orb.py

```diff
@@ -1,197 +1,140 @@
+            "product_name__startswith": platform,
+            "validity_start__lt": start_time.strftime('%Y-%m-%dT%H:%M:%S'),
+            "validity_stop__gt": end_time.strftime('%Y-%m-%dT%H:%M:%S'),
+            "ordering": "-creation_date",
+            "page_size": "1",
+        }
+
+        response = requests.get(url='https://qc.sentinel1.eo.esa.int/api/v1/', params=params)
+        response.raise_for_status()
+        qc_data = response.json()
+
+        orbit_url = None
+        if qc_data["results"]:
+            orbit_url = qc_data["results"][0]["remote_url"]
+        return orbit_url
+
+    elif provider.upper() == 'ASF':
+        url = f'https://s1qc.asf.alaska.edu/{orbit_type.lower()}/'
+        granule = os.path.basename(granule).rstrip('/')
+
+        platform = granule[0:3]
+
+        orbit_url = _get_asf_orbit_url(url, platform, time_stamps[0].replace('T', ''))
+        return orbit_url
+
     else:
-      return None
-  else:
-    print("Using existing orbit file; provider unknown")
-    return stateVecFile
-
-
-def downloadSentinelOrbitFile(granule,directory=None):
-    try:
-        stateVecFile=downloadSentinelOrbitFileProvider(granule,'ASF',directory)
-        provider = "ASF"
-        print("Found state vector file at ASF")
-    except:
-        print("Unable to find statevector at ASF; trying ESA")
+        raise OrbitDownloadError(f'Unkown orbit file provider {provider}')
+
+
+def download_sentinel_orbit_file(granule: str, directory: str = None, providers=('ESA', 'ASF')):
+    """Download a Sentinel-1 Orbit file
+
+    Args:
+        granule: Granule name to find an orbit file for
+        directory: Directory to save the orbit files into
+        providers: Iterable of providers to attempt to download the orbit file from, in order of preference
+
+    Returns: Tuple of:
+        orbit_file: The downloaded orbit file
+        provider: The provider used to download the orbit file from
+
+    """
+    for provider in providers:
+        orbit_url = get_orbit_url(granule, 'AUX_POEORB', provider=provider)
+        if not orbit_url:
+            orbit_url = get_orbit_url(granule, 'AUX_RESORB', provider=provider)
+
+        if not orbit_url:
+            break
+
+        orbit_file = download_file(
+            orbit_url, directory=directory, headers={'User-Agent': 'python3 asfdaac/apt-insar'}, chunk_size=5242880
+        )
+
         try:
-            stateVecFile = downloadSentinelOrbitFileProvider(granule,'ESA',directory) 
-            if stateVecFile:
-                provider = "ESA"
-                print("Found state vector file at ESA")
-        except:
-            print("Unable to find requested state vector file")
-            provider = "NA"
-            return None,provider
-    verify_opod(stateVecFile)
-    return stateVecFile, provider
+            verify_opod(orbit_file)
+        except ValueError:
+            raise OrbitDownloadError(f'Downloaded orbit file is invalid: {orbit_file}')
```

Comment:
if `download_file` or `verify_opod` throw, should we log a warning and advance to the next provider?

Reply:
Hmm, good point. I'll try to address this.